app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/SyncAccountsWithConnectorExtensionUseCase.kt

```diff
@@ -0,0 +1,119 @@
+            .filter { connectionStatuses ->
+                connectionStatuses.isNotEmpty() &&
+                    !connectionStatuses.any { it.value == PeerConnectionStatus.CONNECTING }
+            }
+            .map { statuses ->
+                val generalPurposeConnectionIds = p2pLinksRepository.getP2PLinks()
+                    .filter { it.purpose == P2PLinkPurpose.General }
+                    .map { it.connectionPassword.hexToBagOfBytes().hash().hex }
+                val openConnectionIds = statuses.filter { it.value == PeerConnectionStatus.OPEN }.keys
+
+                generalPurposeConnectionIds.filter { connectionId -> connectionId in openConnectionIds }
+            }
+            .filter { connectionIds -> connectionIds.isNotEmpty() }
+
+        val accountListMessageFlow = getProfileUseCase.activeAccountsOnCurrentNetwork.map { accounts ->
+            val accountListExchangeInteraction = ConnectorExtensionExchangeInteraction.AccountList(
+                accounts = accounts.map { account ->
+                    Account(
+                        address = account.address,
+                        label = account.displayName,
+                        appearanceId = account.appearanceID
+                    )
+                }
+            )
+            json.encodeToString<ConnectorExtensionExchangeInteraction>(accountListExchangeInteraction)
+        }
+
+        return combine(connectionIdsFlow, accountListMessageFlow) { connectionIds, accountListMessage ->
+            val messageHash = accountListMessage.encodeToByteArray()
+                .hash()
+                .hex
+            val lastSyncedMessageHash = preferencesManager.lastSyncedAccountsWithCE.firstOrNull()
+                ?.decodeHex()
+                ?.toHexString()
+
+            if (messageHash != lastSyncedMessageHash) {
+                Timber.d("Accounts sync with CE is required")
+
+                sendAccountListMessage(connectionIds, accountListMessage)
+                    .fold(
+                        onSuccess = {
+                            Timber.d("Successfully synced accounts with CE")
+                            preferencesManager.updateLastSyncedAccountsWithCE(messageHash)
+                        },
+                        onFailure = { throwable ->
+                            Timber.e("Failed to sync accounts with CE. Error: ${throwable.message}")
+                        }
+                    )
+            } else {
+                Timber.d("Accounts sync triggered, but nothing changed since last sync, ignoring..")
+            }
+        }.flowOn(ioDispatcher)
+    }
+
+    private suspend fun sendAccountListMessage(connectionIds: List<String>, message: String): Result<Unit> {
+        return withContext(ioDispatcher) {
+            val jobs = connectionIds.map { connectionId ->
+                async { peerdroidClient.sendMessage(connectionId, message) }
+            }
```

Comment:
I think we should avoid this, because I am not sure how the WebRTC would handle the parallel message dispatching.

Reply:
Sure, I've updated the code to send the messages sequentially. 
FYI: We have the parallel messages sending also in`PeerdroidConnector.sendDataChannelMessageToAllRemoteClients` (we might want to look if there are any problems regarding this there).

Reply:
right, and I realized this very late. We will definitely check and most probably change both.